content/reference/sql/access-management.md

```diff
@@ -0,0 +1,289 @@
+      "Locked": true,
+      "Attributes": null,
+      "IsRole": true
+    },
+    {
+      "User": "test_user",
+      "Host": "localhost",
+      "PrivilegeSet": {
+        "GlobalStatic": [],
+        "Databases": [
+          {
+            "Name": "some_db",
+            "Privileges": [],
+            "Tables": [
+              {
+                "Name": "some_tbl",
+                "Privileges": [
+                  "DROP"
+                ],
+                "Columns": []
+              }
+            ]
+          },
+          {
+            "Name": "other_db",
+            "Privileges": [
+              "CREATE",
+              "DROP"
+            ],
+            "Tables": []
+          }
+        ]
+      },
+      "Plugin": "mysql_native_password",
+      "Password": "",
+      "PasswordLastChanged": "2022-02-15T00:00:00.0000000Z",
+      "Locked": false,
+      "Attributes": null,
+      "IsRole": false
+    }
+  ],
+  "Roles": [
+    {
+      "FromHost": "%",
+      "FromUser": "test_role",
+      "ToHost": "localhost",
+      "ToUser": "test_user",
+      "WithAdminOption": false
+    }
+  ]
+}
+```
+
+## Updates and Persistence
+
+Unlike in MySQL, Dolt immediately processes all updates to the grant tables.
+Any privilege changes will immediately take effect, any deleted users with current sessions will immediately lose access, etc.
+Whether these are benefits or drawbacks depend on those running the database.
+For Dolt, the decision to have all updates take immediate effect allows for emergency updates to not require a server restart in some cases, which we believe offers some security and convenience advantages.
+The benefit of delayed updates do not seem as likely nor often, although we may still change this behavior in the future if it is proven otherwise.
+
+Persistence to the [privilege file](#privilege-file) is immediate only when the grant tables are modified through their typical statements (`CREATE USER`, `GRANT`, `REVOKE`, etc.).
+Directly modifying the grant tables using `INSERT`, `UPDATE`, etc. will cause an immediate update to all current users, however it **will not** be immediately persisted to the [privilege file](#privilege-file).
+The [privilege file](#privilege-file) is **only** updated when the aforementioned statements are executed.
+This may change in the future.
+
+## Statements
+
+For now, only some of the core statements are supported for users and privileges.
+Of those core statements, some are fully supported, while others only offer partial support.
+
+### Fully Supported
+
+- `CREATE ROLE`
+- `DROP ROLE`
+- `DROP USER`
+- `SHOW PRIVILEGES`
+
+The following grant tables are fully implemented:
+
+- `mysql.user`
+  - Contains the user definition, global static privileges, login details, password limits, account maximums, and attributes
+  - Although this table is fully implemented, we do not support all of the features that this table provides, even though they may be set
```

Comment:
I'm not sure how hard it would be to log warnings on unsupported config values, but might be worth considering if it's not too tough. It's always so frustrating as a user when you think you've got the syntax correct and you can't figure out why a features isn't working as expected.

Reply:
For some of the scenarios, implementing the work to emit a warning and implementing the feature aren't too different in the amount of time spent. Something to think about going forward though for cases where that's not true.